src/libraries/System.IO.Compression.Zstandard/src/System/IO/Compression/ZstandardDictionary.cs

```diff
@@ -0,0 +1,207 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.IO.Compression
+{
+    /// <summary>Represents a Zstandard compression dictionary.</summary>
+    public sealed class ZstandardDictionary : IDisposable
+    {
+        private readonly SafeZstdCDictHandle _compressionDictionary;
+        private readonly SafeZstdDDictHandle _decompressionDictionary;
+        private readonly byte[] _dictionaryData;
+        private bool _disposed;
+
+        private ZstandardDictionary(SafeZstdCDictHandle compressionDict, SafeZstdDDictHandle decompressionDict, byte[] data)
+        {
+            _compressionDictionary = compressionDict;
+            _decompressionDictionary = decompressionDict;
+            _dictionaryData = data;
```

Comment:
Could anything get corrupted / any security/safety issues / etc. occur if the dictionary's data was changed concurrently with its use? This is stored the raw byte[] passed in, which the user will still have access to.